### DISCUSSION TITLE
Unexpected hist_equal result

### DISCUSSION THREAD

**Discussion by Nakilon** [libvips/ruby-vips#307] [Q&A] [answered]
I have such image:
![2](https://user-images.githubusercontent.com/2870363/121788090-238e5300-cbd3-11eb-9392-1bc931baff0a.png)

I process it like this and see that it has been splitted into 14 regions:
```
img.>(65).labelregions.to_a.flatten.uniq
[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13]
```

Since they are dark, to see them I apply the `hist_equal`, but instead of seeing something like this:
![image](https://user-images.githubusercontent.com/2870363/121788418-d364c000-cbd5-11eb-81d4-d7a29b44b20b.png)

It drops several colors:
```
img.>(65).labelregions.hist_equal.to_a.flatten.uniq
[9, 10, 11, 12]
```
![image](https://user-images.githubusercontent.com/2870363/121788491-4a01bd80-cbd6-11eb-9c8f-043d43f82ce7.png)

Is it correct? Is there something I don't understand?

**Comment by jcupitt** [comment, marked as answer]
Hi @Nakilon,

I think that's correct -- the normalization step in equalization can join very nearby values together.

I usually scale and falsecolour to visualize region labelleling, for example:

```
x = Vips::Image.new_from_file("regions.png")
labelled, opts = x.labelregions(segments: TRUE)
vis = (labelled * 256 / opts["segments"]).falsecolour
```

To make:

![y](https://user-images.githubusercontent.com/580843/121860244-92ad9980-ccf0-11eb-9411-50aa077de161.png)

nip2 has a button to do this (very handy).

**Reply by Nakilon** [reply]
Cool. Didn't know about `falsecolor`.
